### ISSUE TITLE
Regression: Usage of VK_KHR_display produces false positives since v1.1.123

### ISSUE THREAD

**Issue by lubosz** [KhronosGroup/Vulkan-ValidationLayers#1660]
Running the Monado compositor with recent Vulkan validation introduces new errors I am unable to get rid of with API usage.
I am using the `VK_KHR_display` extension in combination with `VK_EXT_direct_mode_display` and `VK_EXT_acquire_xlib_display`. Not sure if the latter are related to the problem since I don't have a more minimal `VK_KHR_display` example. Therefore I also did not test this behavior on other platforms than Xlib.

This code runs correctly on 3 different Vulkan implementations, but produces the same validation errors on all 3 of them. The tested implementations were:
* radv, the open source AMD driver from Mesa (20.0.2)
* amdvlk: AMD's new stand alone open source driver (2020.Q1.3)
* Intel's Open Source Vulkan driver from Mesa (20.0.2)

The problem occurs on Vulkan loader and headers at `v1.2.135`.

I wrote an executable minimal 250 line example that is able to reproduce the errors.
https://gitlab.freedesktop.org/lubosz/vulkan-display-test/-/blob/master/main.c

Instructions on how to build and run the test can be found in it's README.

All Vulkan API calls in this example are validated with `assert(res == VK_SUCCESS);` and it builds in clang and gcc without warnings on a high warning level.

This example does not produce related errors in valgrind, it runs cleanly besides 2 errors in LLVM, which are probably produced by radv / mesa.

I was able to find the commits that introduced the errors using `git bisect`.

## Couldn't find Object

The first error states `Couldn't find VkDisplayKHR Object`, so the handle seems to be unknown to the validation. The code is working correctly though.

The error is caused by all functions taking the handle as a parameter, being the `vkAcquireXlibDisplayEXT` and `vkReleaseDisplayEXT` calls, as well as `vkGetDisplayModePropertiesKHR`.
The handle is produced by the `vkGetRandROutputDisplayEXT` function.

```
UNASSIGNED-Threading-Info(ERROR / SPEC): msgNum: 0 - [ UNASSIGNED-Threading-Info ] Object: 0x10000000001 (Type = 1000002000) | Couldn't find VkDisplayKHR Object 0x10000000001. This should not happen and may indicate a bug in the application.
    Objects: 1
        [0] 0x10000000001, type: 1000002000, name: NULL
```

The error was introduced by 355b7ed5e73b8cb1efdadddf5ece28ee8af60c9c, first tagged in `v1.1.123` or `sdk-1.1.121.0` (Don't know the difference between the regular version and the SDK tags).
```
355b7ed5e73b8cb1efdadddf5ece28ee8af60c9c is the first bad commit
commit 355b7ed5e73b8cb1efdadddf5ece28ee8af60c9c
Author: Jeff Bolz <jbolz@nvidia.com>
Date:   Thu Aug 22 12:09:54 2019 -0500

    layers: Use persistent objects for ThreadSafety tracking
    
    The previous 'Bucket' change was a big improvement, but it has a
    significant remaining problem that two threads *reading* the same
    object will fight over a *write* lock for the same bucket. This change
    creates a tracking object (ObjectUseData) that persists for the
    lifetime of the object and is manipulated using atomic adds to track
    the read and write counts. The object is looked up from a
    concurrent_unordered_map, which only needs a read lock so two threads
    each reading won't collide.
    
    Change-Id: I49f429d93efa7f86ee8d8c4159dabaf6353ae22e
```

The funny thing is that the validation detected this object as `VkSwapchainKHR` type until quite recently. Until commit 8c312ef4c7a95427f337a4be000c7191c6705126 tagged only in `v1.2.135` the type is correctly detected as `VkDisplayKHR`.

```
UNASSIGNED-Threading-Info(ERROR / SPEC): msgNum: 0 - Couldn't find VkSwapchainKHR Object 0x10000000001. This should not happen and may indicate a bug in the application.
    Objects: 1
        [0] 0x10000000001, type: 1000002000, name: NULL
```

For the following reasons I suspec this validation error is a false positive:
* The application is able to work with the `VkDisplayKHR` correctly.

## Object has not been destroyed

```
VUID-vkDestroyInstance-instance-00629(ERROR / SPEC): msgNum: 0 - [ VUID-vkDestroyInstance-instance-00629 ] Object: 0x555c9f1392a0 (Type = 1) | OBJ ERROR : For VkInstance 0x555c9f1392a0[], VkDisplayKHR 0x10000000001[] has not been destroyed. The Vulkan spec states: All child objects created using instance must have been destroyed prior to destroying instance (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyInstance-instance-00629)
    Objects: 1
        [0] 0x555c9f1392a0, type: 1, name: NULL
```

This leak occurs even though the display is released via the `vkReleaseDisplayEXT` function.

Especilly this error is quite verbose, since it is printed for every mode enumerated:
```
VUID-vkDestroyInstance-instance-00629(ERROR / SPEC): msgNum: 0 - [ VUID-vkDestroyInstance-instance-00629 ] Object: 0x555c9f1392a0 (Type = 1) | OBJ ERROR : For VkInstance 0x555c9f1392a0[], VkDisplayModeKHR 0x20000000002[] has not been destroyed. The Vulkan spec states: All child objects created using instance must have been destroyed prior to destroying instance (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyInstance-instance-00629)
    Objects: 1
        [0] 0x555c9f1392a0, type: 1, name: NULL

```

The leak errors were introduced in 7cef2637ff3ac3f5a11a5328f679bf1d138f7105 and released with v1.1.123 (sdk-1.1.121.0).
```
7cef2637ff3ac3f5a11a5328f679bf1d138f7105 is the first bad commit
commit 7cef2637ff3ac3f5a11a5328f679bf1d138f7105
Author: Mark Lobodzinski <mark@lunarg.com>
Date:   Wed Aug 28 15:50:13 2019 -0600

    objlifetimes: Fix up instance/device undestroyed obj reporting
    
    No distinction existed between objects having instance or device
    parent types, necessary for properly reporting the correct VUID.
    
    Change-Id: If38f36517693f4d2b2ae69a630c5437b59e4c344
```

For the following reasons I suspec this validation errors are a false positive:
* valgrind does not report anything related to the leaks
* The `VkDisplayModeKHR` leaks are caused by simply running `vkGetDisplayModePropertiesKHR` with a non NULL `VkDisplayPropertiesKHR* pProperties` parameter
* `VkDisplayModeKHR` does not have any destructor in the API
* `vkReleaseDisplayEXT` is called correctly and returns without an error.

**Comment by lubosz** [comment]
Thanks, can confirm it's fixed on today's git master.

**Comment by mark-lunarg** [comment]
@lubosz, thanks for the verification, and I wanted to especially thank you for the detailed issue report.  It should be a model for all submitters!  We *really* appreciate it.
